driver/src/db_interface.rs

```diff
@@ -0,0 +1,137 @@
+pub mod db_interface{    
+    use mongodb::bson;
+    use mongodb::db::ThreadedDatabase;
+    use mongodb::{Client, ThreadedClient};
+
+    extern crate models;
+    extern crate rustc_hex;
+    extern crate web3;
+
+    use web3::types::H256;
+
+    use std::io;
+    use std::io::{Error, ErrorKind};
+    
+    #[derive(Clone)]
+    pub struct DbInterface{
+        pub client: Client,
+        pub db_host: String,
+        pub db_port: String,
+    }
+
+    impl DbInterface {
+        pub fn new(db_host: String, db_port: String) -> Result<DbInterface, &'static str> {
+            let client = Client::connect(&db_host, db_port.parse::<u16>().unwrap())
+                .expect("Failed to initialize standalone client");
+
+            Ok(DbInterface { client, db_host, db_port })
+        }
+
+        pub fn get_current_balances(&self, current_state_root: H256) -> Result<models::State, Error> {
+            let t: String = format!("{:#x}", current_state_root);
+            let mut query = String::from(r#" { "stateHash": ""#);
```

Comment:
nit: maybe construct the query with one `format!`

Reply:
Yes, this would be nice. however, I was not able to process a raw string with format!